lib/src/services/preferences.dart

```diff
@@ -30,6 +33,19 @@ class Preferences extends Service {
 		return result;
 	}
 
+	/// Returns the last date the data was retrieved from the
+	/// database as [DateTime].
+	DateTime? getRefreshData(String data){
+		final String? lastUpdate = _prefs.getString(refreshData(data));
+				return DateTime.parse(lastUpdate!);
+	}
+
+	/// Saves the Date when the data was last retrieved from
+	/// the database as type [String].
+
+	Future<void> setRefreshData(String data, DateTime value) async{
+		await _prefs.setString(refreshData(data), value.toString());
+	}
```

Comment:
1. Something I figured out quite late is that you actually don't need `async`/`await` -- you can return the `Future<void>` given to you by `_prefs.setString` directly. 
2. If you know you're always going to use `DateTime.now`, you might as well use it here. 
3. You shouldn't name it `data` because it doesn't actually hold any data, just the _name_ of the data. You shouldn't name it `key` either, since the actual key is `refresh_X`. `name seems to be better. 
4. You might as well get rid of the above `refreshData` function and make the string directly. 

```suggestion
	Future<void> setRefreshData(String name) => 
		_prefs.setString("refresh_$name", DateTime.now().toString());
```